store/rootmulti/store.go

```diff
@@ -475,19 +475,20 @@ func (rs *Store) CacheWrapWithTrace(_ io.Writer, _ types.TraceContext) types.Cac
 	return rs.CacheWrap()
 }
 
-// CacheWrapWithListeners implements the CacheWrapper interface.
-func (rs *Store) CacheWrapWithListeners(_ types.StoreKey, _ []types.WriteListener) types.CacheWrap {
-	return rs.CacheWrap()
-}
-
 // CacheMultiStore creates ephemeral branch of the multi-store and returns a CacheMultiStore.
 // It implements the MultiStore interface.
 func (rs *Store) CacheMultiStore() types.CacheMultiStore {
 	stores := make(map[types.StoreKey]types.CacheWrapper)
 	for k, v := range rs.stores {
-		stores[k] = v
+		store := v.(types.KVStore)
```

Comment:
What happens when `v` is not a `types.KVStore`?

Reply:
the `v` is `types.CommitKVStore` which is a superset of `types.KVStore`.

Reply:
In that case there's no reason to make a type assertion, you can just use `v` directly instead of `store`, right?

Reply:
looks much better, I also changed a few similar cases, thanks for the suggestion.